app/src/main/java/com/github/featuredetectandroid/utils/KeypointExtractionAlgorithm.kt

```diff
@@ -0,0 +1,31 @@
+package com.github.featuredetectandroid.utils
+
+import android.util.Log
+
+private const val TAG = "KeypointExtractionAlgorithm"
+
+enum class KeypointExtractionAlgorithm {
+    NONE,
+    SIFT,
+    SURF,
+    ORB,
+    SUPERPOINT
+}
+
+fun algorithmsNamesAsList() = enumValues<KeypointExtractionAlgorithm>().map {
+    when (it) {
+        KeypointExtractionAlgorithm.NONE -> "None"
+        KeypointExtractionAlgorithm.SUPERPOINT -> "SuperPoint"
+        else -> it.name
+    }
+}
+
+fun getAlgorithmByName(name: String): KeypointExtractionAlgorithm {
```

Comment:
Seems like it's not used anywhere

Reply:
> Seems like it's not used anywhere

This is a stub to use to pass the selection of algorithm to the keypoint extraction function.

Reply:
I think, you're using the word "stub" wrong here. Stub it something that does nothing and will be replaced later. And this is a function that you'll need to use later, do I understand you right?

Reply:
> I think, you're using the word "stub" wrong here. Stub it something that does nothing and will be replaced later. And this is a function that you'll need to use later, do I understand you right?

Yes, I edited the comment. I meant the function that is unused for now, but is going to be used in the future.